### ISSUE TITLE
Unable to build ghc-lib 8.8.* from ghc 8.6

### ISSUE THREAD

**Issue by zliu41** [digital-asset/ghc-lib#309]
There are a bunch of errors like this when building ghc-lib 8.8 using ghc 8.6:
```
dist/build/CmmParse.hs:3384:24: error:
    Not in scope: data constructor ‘Prelude.False’
    Perhaps you meant ‘GhcPrelude.False’ (imported from GhcPrelude)
    No module named ‘Prelude’ is imported.
```

**Comment by shayne-fletcher** [comment]
we've seen this before. the function `applyPatchCmmParseNoImplicitPrelude` exists to fix it. it's mysterious as to why it's reproducibility has been so haphazard but in short for whatever reason `CmmParse.y` is missing a directive `import qualified Prelude`. the fix would be to cut new 8.8.* releases.

**Comment by shayne-fletcher** [comment]
the issue where this was first reported: https://github.com/digital-asset/ghc-lib/issues/243

**Comment by shayne-fletcher** [comment]
@zliu41 i've released 8.8.1.20210620. can you please advise on whether it resolves this issue?

**Comment by zliu41** [comment]
Yeah this fixes it. Thanks for the quick resolution!
